app/json_convert/src/main.cpp

```diff
@@ -0,0 +1,188 @@
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <csignal>
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+#include "lib/dls/dls.h"
+#include "lib/telemetry/TelemetryViewer.h"
+#include "lib/vcm/vcm.h"
+
+#define NUM_SIGNALS 5
+#define SERVER_IP "127.0.0.1"
+#define SERVER_PORT 5000
+
+
+using namespace dls;
+using namespace vcm;
+
+TelemetryViewer tlm;
+
+bool killed = false;
+
+const int SIGNALS[NUM_SIGNALS] = {SIGINT, SIGTERM, SIGSEGV, SIGFPE, SIGABRT};
+
+void err_handle(std::string message, MsgLogger *logger) {
+    std::string err_msg = "JSON_Convert: ";
+    err_msg.append(message);
+
+    logger->log_message(err_msg);
+    std::cout << err_msg << std::endl;
+
+    exit(-1);
+}
+
+void sighandler(int) {
+    killed = true;
+    tlm.sighandler();
+}
+
+std::string getJSONString(VCM *vcm, MsgLogger *logger, size_t max_size) {
+    std::string jsonString = "{";
+
+    for (std::string measurement : vcm->measurements) {
```

Comment:
TelemetryViewer::update needs to be called in order to see new telemetry data, likely call it in this function? TelemetryViewer keeps a local cache of measurements and refreshes it when 'update' is called. In your case you set the update mode to blocking so it will block until new telemetry has been received.